simtools/simtel/simtel_histograms.py

```diff
@@ -67,50 +69,184 @@ def get_histogram_title(self, i_hist):
+        Raises an error in case they are not consistent.
+
+        Parameters
+        ----------
+        first_hist_file: dict
+            One histogram from a single file.
+        second_hist_file: dict
+            One histogram from a single file.
+
+        Raises
+        ------
+        BadHistogramFormat:
+            if the format of the histograms have inconsistent dimensions.
+        """
+        for key_to_test in [
+            "lower_x",
+            "upper_x",
+            "n_bins_x",
+            "title",
+        ]:
+            if first_hist_file[key_to_test] != second_hist_file[key_to_test]:
+                msg = "Trying to add histograms with inconsistent dimensions"
+                self._logger.error(msg)
+                raise BadHistogramFormat(msg)
+
+    def combine_histogram_files(self):
+        """Add the values of the same type of histogram from the various lists into a single
+        histogram list."""
+        # Processing and combining histograms from multiple files
+        self.combined_hists = []
+        n_files = 0
+        for hists_one_file in self.list_of_histograms:
+            count_file = True
+
+            if len(self.combined_hists) == 0:
+                # First file
+                self.combined_hists = copy.copy(hists_one_file)
+
+            else:
+                for hist, this_combined_hist in zip(hists_one_file, self.combined_hists):
+                    self._check_consistency(hist, this_combined_hist)
 
-                    if len(self.combined_hists) == 0:
-                        # First file
-                        self.combined_hists = copy.copy(hists)
-
-                    else:
-                        # Remaining files
-                        for hist, this_combined_hist in zip(hists, self.combined_hists):
-                            # Checking consistency of histograms
-                            for key_to_test in [
-                                "lower_x",
-                                "upper_x",
-                                "n_bins_x",
-                                "title",
-                            ]:
-                                if hist[key_to_test] != this_combined_hist[key_to_test]:
-                                    msg = "Trying to add histograms with inconsistent dimensions"
-                                    self._logger.error(msg)
-                                    raise BadHistogramFormat(msg)
-
-                            this_combined_hist["data"] = np.add(
-                                this_combined_hist["data"], hist["data"]
-                            )
-
-                    n_files += int(count_file)
+                    this_combined_hist["data"] = np.add(this_combined_hist["data"], hist["data"])
+
+            n_files += int(count_file)
 
         self._logger.debug(f"End of reading {n_files} files")
 
+    def _derive_trigger_rate_histograms(self, livetime=1):
```

Comment:
Can you help with with the default value for the livetime of 1 s. Why? Seems also that _derive_trigger_rate_histograms is called always with livetime set, so maybe this is not necessary?

Reply:
Hmm, it doesn't really need to be one as default. I am making it a required input for now but I am going to check whether we can get it directly from the histogram file (therefore I am leaving this conversation open)

Reply:
As far as I could find out, there is no such thing as "livetime" in the simtel_array output files. That is understandable, as the simulations are conducted based on the number of events and not on livetime and some reference flux. Therefore, we will leave as it is now and implement the remaining features in the next steps.

Reply:
But is this a reasonable default? 1s seems to be weird, why not simply having no default and call the function `def _derive_trigger_rate_histograms(self, livetime)`? It is quite reasonable to request that lifetime must be given as an argument when calling this function.

Reply:
OK, this is good! Sorry, didn't see that - it would really be nice if the github interface would indicate also changes to lines one has commented on. So all good!!

Reply:
agreed!